topics/basics/testing_plugins/integration_tests/integration_tests_api.md

```diff
@@ -0,0 +1,197 @@
+3. When the client calls a method on the stub:
+    * The stub serializes the method call, including any parameters.
+    * It sends the serialized data over the network to the remote server.
+    * The RMI runtime inside the IDE unmarshals the request and invokes the corresponding method on the actual remote object.
+    * The method executes, and the result is serialized and sent back to the client via the stub.
+    * The stub unmarshals the response and returns the result to the caller.
+
+Thus, the stub makes remote calls feel like local method calls, without the caller worrying about low-level networking, data conversion, or request handling.
+
+## Creating stubs
+
+To demonstrate how this works in practice, add the following code to the plugin:
+
+```kotlin
+class PluginStorage {
+  companion object {
+    @JvmStatic
+    fun getPluginStorage() = Storage("static method", listOf("static1", "static2"))
+  }
+}
+
+@Service
+class PluginService {
+  fun someMethod(): Unit = Unit
+  fun getAnswer(): Int = 42
+}
+
+@Service(Service.Level.PROJECT)
+class PluginProjectService {
+  fun getStrings(): Array<String> = arrayOf("foo", "bar")
+}
+
+data class Storage(val key: String, val attributes: List<String>)
+```
+
+This file contains one class with the static method `getPluginStorage`, and two [light services](plugin_services.md#light_services) – one application-level `PluginService` and one project-level `PluginProjectService`.
+
+The test needs to call the methods of these classes and verify their return values.
+To do this, the following stubs have to be created.
+
+```kotlin
+import com.intellij.driver.client.Remote
+
+@Remote("org.jetbrains.testPlugin.PluginStorage", plugin = "org.example.demo")
+interface PluginStorage {
+  fun getPluginStorage(): Storage
+}
+
+@Remote("org.jetbrains.testPlugin.PluginService", plugin = "org.example.demo")
+interface PluginService {
+  fun getAnswer(): Int
+}
+
+@Remote("org.jetbrains.testPlugin.PluginProjectService", plugin = "org.example.demo")
+interface PluginProjectService {
+  fun getStrings(): Array<String>
+}
+
+@Remote("org.jetbrains.testPlugin.Storage", plugin = "org.example.demo")
+interface Storage {
+  fun getAttributes(): List<String>
+  fun getKey(): String
+}
+```
+
+As seen above, a stub is an interface for the class with methods that will be used in the test.
+Stubs shouldn't be created for methods not used in tests.
+
+The first parameter is the fully qualified name of the class that will correspond to the stub using the [`Remote`](%gh-ic%/platform/remote-driver/client/src/com/intellij/driver/client/Remote.kt) annotation.
+Strings are used to avoid introducing dependency between production and test code.
+
+The second parameter specifies `pluginId`, where classes are located.
+This parameter is required since IntelliJ-based IDEs use separate classloaders for each plugin, and the code that will call methods on the IDE side (Invoker) needs to know where to search for them.
```

Comment:
```suggestion
This parameter is required since IntelliJ-based IDEs use separate [class loaders](plugin_class_loaders.md) for each plugin, and the code that will call methods on the IDE side (Invoker) needs to know where to search for them.
```